src/OpenLoco/src/Audio/Audio.cpp

```diff
@@ -1149,11 +1174,6 @@ namespace OpenLoco::Audio
             return;
         }
 
-        if (currentTrackPathId == sample)
```

Comment:
This removes the only use of `currentTrackPathId` all others are just writing to it. Are you sure this makes sense.

Reply:
This check is what causes the bug, as it prevents the same track from playing twice in a row.

Other than preventing the same track from playing twice in a row, which in this pull request is instead handled by setting the exclude track to the previous track, I cannot figure out what situation this check prevents. The commits and pull request that added it does not explain why it was added.

So that is my reasoning for why I removed it rather than finding some way to circumvent its bug-causing behaviour